File: 🟢 Easy/1945 - Sum of Digits of String After Convert/_1945SumofDigitsofStringAfterConvert.cpp
class Solution {
public:
    string transform(string str) {
        int ans = 0;
        for (int i = 0; i < str.length(); i++) {
            ans += str.at(i) - '0';
        }
        return to_string(ans);
    }

    int getLucky(string s, int k) {
        string convertedVal = "";
        for (int i = 0; i < s.length(); i++) {
            char ch = s.at(i);
            int n = (ch - 'a') + 1;
            convertedVal = convertedVal + to_string(n);
        }

        while (k > 0) {
            convertedVal = transform(convertedVal);
            k--;
        }

        return stoi(convertedVal);
    }
};

File: 🟢 Easy/1945 - Sum of Digits of String After Convert/_1945SumofDigitsofStringAfterConvert.js
/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */

function transform(str) {
    let ans = 0;
    for (let i = 0; i < str.length; i++) {
        ans += Number(str[i]);
    }
    return ans.toString();
}

var getLucky = function (s, k) {
    let convertedVal = "";
    for (let i = 0; i < s.length; i++) {
        let n = (s.charCodeAt(i) - 97) + 1;
        convertedVal = convertedVal + n.toString();
    }

    while (k > 0) {
        convertedVal = transform(convertedVal);
        k--;
    }

    return parseInt(convertedVal);
};